### PR TITLE
Make resteasy execute rpc filter chain too.

### LINKED ISSUE #376: 自定义 filter 对于 rest 协议不起作用
### Describe the bug

自定义 filter 对于 rest 协议不起作用

 RestServer 180 行: 

```java
  @Override
    public void registerProcessor(ProviderConfig providerConfig, Invoker instance) {
        if (!isStarted()) {
            start();
        }
        // 在httpserver中注册此jaxrs服务
        if (LOGGER.isInfoEnabled()) {
            LOGGER.info("Register jaxrs service to base url http://" + serverConfig.getHost() + ":"
                + serverConfig.getPort() + serverConfig.getContextPath());
        }
        try {
            httpServer.getDeployment().getRegistry()
                .addResourceFactory(new SofaResourceFactory(providerConfig), serverConfig.getContextPath());

            invokerCnt.incrementAndGet();
        } catch (Exception e) {
            LOGGER.error("Register jaxrs service error", e);
        }
    }
```

参数 `Invoker instance` 没有用到,导致 invoker 外面包的 filterChain 没有用到.
SofaResourceFactory 中:

```java
public SofaResourceFactory(ProviderConfig providerConfig) {
        super(providerConfig.getRef());
        this.providerConfig = providerConfig;
        // 缓存服务名计算和应用名计算
        this.serviceName = ConfigUniqueNameGenerator.getServiceName(providerConfig);
        this.appName = providerConfig.getAppName();
    }
```
应该新建一个代理来替代`providerConfig.getRef()`

### Environment

- SOFARPC version:5.4.7
- JVM version (e.g. `java -version`):
- OS version (e.g. `uname -a`):
- Maven version:
- IDE version:

### PR DESCRIPTION
**Pull request by OrezzerO**
### Motivation:

fix #376

### Modification:

fix #376

### Result:

Fixes #376

### CODE REVIEW

File: test/test-integration/src/test/java/com/alipay/sofa/rpc/server/rest/CustomizeFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    static CustomizeTestFilter         providerFilter;
+
+    static CustomizeTestFilter         clientFilter;
+    static ProviderConfig<RestService> providerConfig;
+
+    static {
+        JAXRSProviderManager.registerCustomProviderInstance(new CustomizeContainerRequestTestFilter());
+        JAXRSProviderManager.registerCustomProviderInstance(new CustomizeContainerResponseTestFilter());
+        JAXRSProviderManager.registerCustomProviderInstance(new CustomizeClientRequestTestFilter());
+        JAXRSProviderManager.registerCustomProviderInstance(new CustomizeClientResponseTestFilter());
```
Comment (ujjboy) [inline test/test-integration/src/test/java/com/alipay/sofa/rpc/server/rest/CustomizeFilterTest.java]: use `@beforeClass` method for `registerCustomProviderInstance()`, and unregister them by `@afterClass` method.

File: test/test-integration/src/test/java/com/alipay/sofa/rpc/server/rest/CustomizeFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            .setApplication(new ApplicationConfig().setAppName("TestClientRest"));
+        filterRestService = consumerConfigRest.refer();
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        JAXRSProviderManager.registerCustomProviderInstance(customizeContainerRequestTestFilter);
+        JAXRSProviderManager.registerCustomProviderInstance(customizeContainerResponseTestFilter);
+        JAXRSProviderManager.registerCustomProviderInstance(customizeClientRequestTestFilter);
+        JAXRSProviderManager.registerCustomProviderInstance(customizeClientResponseTestFilter);
```
Comment (ujjboy) [inline test/test-integration/src/test/java/com/alipay/sofa/rpc/server/rest/CustomizeFilterTest.java]: Need call `removeCustomProviderInstance` after run test case, not `registerCustomProviderInstance`

Comment (OrezzerO) [reply test/test-integration/src/test/java/com/alipay/sofa/rpc/server/rest/CustomizeFilterTest.java]: my fault

**Comment by OrezzerO** [comment]
CI failed ,how to rerun it ?

**Comment by leizhiyuan** [comment]
> CI failed ,how to rerun it ?

I will do it again。

**Pull request merged**